src/structures/Message.js

```diff
@@ -584,7 +584,10 @@ class Message extends Base {
    */
   get editable() {
     const precheck = Boolean(
-      this.author.id === this.client.user.id && !deletedMessages.has(this) && (!this.guild || this.channel?.viewable),
+      this.author.id === this.client.user.id &&
+        !deletedMessages.has(this) &&
+        !this.flags.has(MessageFlags.FLAGS.EPHEMERAL) &&
```

Comment:
Ephemeral messages can be edited

Reply:
not via `Message#edit` though

Reply:
one can argue either way, imo if you edit ephemeral messages via `Interaction#editReply` you wouldn't use this getter since you don't need any permissions for it but it's up to you maintainers whether to include this check

Reply:
> not via `Message#edit` though

What if we make an `InteractionMessage` that takes the interaction and makes the message editable? 🤔

Would also remove the need to use not-so-intuitive methods.